server/game/core/attack/AttackFlow.ts

```diff
@@ -59,22 +65,32 @@ export class AttackFlow extends BaseStepWithPipeline {
             overwhelmDamageOnly = true;
         }
 
-        let damageEvents: GameEvent[];
-
+        const attackerDealsDamageBeforeDefender = this.attack.attackerDealsDamageBeforeDefender();
         if (overwhelmDamageOnly) {
-            damageEvents = [AbilityHelper.immediateEffects.damage({ amount: this.attack.getAttackerTotalPower() }).generateEvent(this.attack.target.controller.base, this.context)];
+            const damageEvents = [AbilityHelper.immediateEffects.damage({ amount: this.attack.getAttackerTotalPower() }).generateEvent(this.attack.target.controller.base, this.context)];
+            this.context.game.openEventWindow(damageEvents);
```

Comment:
I think this got lost in my previous comment since it ended up spiraling out into a bigger change, but here we could consolidate down to one line which will both create the event and open the window:

```typescript
AbilityHelper.immediateEffects.damage({ amount: this.attack.getAttackerTotalPower() }).resolve(this.attack.target.controller.base, this.context);
```

Reply:
replaced two lines with one line.